framework/console/controllers/MessageController.php

```diff
@@ -133,21 +133,26 @@ public function actionExtract($configFile)
                     $this->saveMessagesToPHP($messages, $dir, $config['overwrite'], $config['removeUnused'], $config['sort']);
                 }
             }
-        } elseif ($config['format'] === 'db') {
-            $db = \Yii::$app->get(isset($config['db']) ? $config['db'] : 'db');
-            if (!$db instanceof \yii\db\Connection) {
-                throw new Exception('The "db" option must refer to a valid database application component.');
+        } elseif (in_array($config['format'], ['db', 'pot'])) {
+            if ($config['format'] === 'db') {
```

Comment:
Ok, changed.
Originaly I kept them together as "pot" and "db" do not need to iterate over the languages